src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpClassification.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpClassification : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpClassification(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpClassification))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void Verify_Color_Of_Some_Tokens()
+        {
+            SetEditorText(@"using System;
+using System.Collections.Generic;
+using System.Text;
+namespace ConsoleApplication1
+{
+    /// <summary>innertext
+    /// </summary>
+    /// <!--comment-->
+    /// <![CDATA[cdata]]>
+    /// <typeparam name=""attribute"" />
+    public class Program
+        {
+            public static void Main(string[] args)
+            {
+                Console.WriteLine(""Hello World"");
+            }
+        }
+    }");
+
+            PlaceCaret("class");
+            VerifyTokenType(tokenType: "keyword");
+            PlaceCaret("{");
+            VerifyTokenType(tokenType: "punctuation");
+            PlaceCaret("Program");
+            VerifyTokenType(tokenType: "class name");
+            PlaceCaret("Main");
+            VerifyTokenType(tokenType: "identifier");
+            PlaceCaret("Hello");
+            VerifyTokenType(tokenType: "string");
+            PlaceCaret("Hello");
+            VerifyTokenType(tokenType: "string");
+            PlaceCaret("<summary", charsOffset: -1);
+            VerifyTokenType(tokenType: "xml doc comment - delimiter");
+            PlaceCaret("summary");
+            VerifyTokenType(tokenType: "xml doc comment - name");
+            PlaceCaret("innertext");
+            VerifyTokenType(tokenType: "xml doc comment - text");
+            PlaceCaret("comment");
+            VerifyTokenType(tokenType: "xml doc comment - comment");
+            PlaceCaret("CDATA");
+            VerifyTokenType(tokenType: "xml doc comment - delimiter");
+            PlaceCaret("cdata");
+            VerifyTokenType(tokenType: "xml doc comment - cdata section");
+            PlaceCaret("attribute");
+            VerifyTokenType(tokenType: "identifier");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void Semantic_Classification()
```

Comment:
It could.  This test is verifying semantic classification and the previous test is verifying parser classification.  It seems reasonable to keep them separate

---
In reply to: [104517159](https://github.com/dotnet/roslyn/pull/17546#discussion_r104517159) [](ancestors = 104517159)